source/base/bounding_box.cc

```diff
@@ -55,42 +52,53 @@ BoundingBox<spacedim, Number>::merge_with(
                  other_bbox.boundary_points.first[i]);
       this->boundary_points.second[i] =
         std::max(this->boundary_points.second[i],
-                 other_bbox.boundary_points.second[i]);
+                    other_bbox.boundary_points.second[i]);
     }
 }
 
-
+template <int spacedim, typename Number>
+bool
+BoundingBox<spacedim, Number>::is_neighbor(
+  const BoundingBox<spacedim, Number> &other_bbox,
+  const double                         tolerance) const
+{
+  for (unsigned int i = 0; i < spacedim; ++i)
+    {
+      // testing if the boxes are close enough to intersect
+      if ((other_bbox.boundary_points.second[i] <
+           this->boundary_points.first[i] - tolerance * side_length(i)) ||
+          (other_bbox.boundary_points.first[i] >
+           this->boundary_points.second[i] + tolerance * side_length(i)))
+        return false;
+    }
+  return true;
+}
 
 template <int spacedim, typename Number>
 NeighborType
 BoundingBox<spacedim, Number>::get_neighbor_type(
   const BoundingBox<spacedim, Number> &other_bbox,
   const double                         tolerance) const
 {
+  if (!is_neighbor(other_bbox, tolerance))
+    return NeighborType::not_neighbors;
+
   if (spacedim == 1)
     {
       // In dimension 1 if the two bounding box are neighbors
       // we can merge them
-      if (this->point_inside(other_bbox.boundary_points.first, tolerance) ||
-          this->point_inside(other_bbox.boundary_points.second, tolerance))
-        return NeighborType::mergeable_neighbors;
-      return NeighborType::not_neighbors;
+      return NeighborType::mergeable_neighbors;
     }
   else
     {
+
       const std::array<Point<spacedim, Number>, 2> bbox1 = {
         {this->get_boundary_points().first,
          this->get_boundary_points().second}};
       const std::array<Point<spacedim, Number>, 2> bbox2 = {
         {other_bbox.get_boundary_points().first,
          other_bbox.get_boundary_points().second}};
 
-      // Step 1: testing if the boxes are close enough to intersect
-      for (unsigned int d = 0; d < spacedim; ++d)
-        if (bbox1[0][d] * (1 - tolerance) > bbox2[1][d] ||
-            bbox2[0][d] * (1 - tolerance) > bbox1[1][d])
-          return NeighborType::not_neighbors;
-
       // The boxes intersect: we need to understand now how they intersect.
       // We begin by computing the intersection:
       std::array<double, spacedim> intersect_bbox_min;
```

Comment:
Same here

Reply:
So you want me to split the introduction of the function and the usage in the code? Is this because it automatically fixes a bug, or should I proceed this way in general?